src/entities/user/user.service.ts

```diff
@@ -43,11 +46,27 @@ class UserService {
         }
     };
 
-    createUser = async (payload: UserAttributes): Promise<string> => {
+    createUser = async (payload: UserAttributes): Promise<string | HttpException> => {
         try {
-            const newUser = this.userModel.build(payload);
+            const { password, login } = payload;
+            const userNameIsUsed = await this.userModel.findOne({ where: { login } });
+            if (!!userNameIsUsed) {
+                Logger.error(`Login ${login} already exists, please try an unique name.`);
+                return new HttpException(
+                    StatusCodes.FORBIDDEN,
+                    `Login ${login} already exists, please try an unique name.`,
+                );
+            }
+            const salt = await bcrypt.genSalt();
```

Comment:
Consider creating a class/object in services/utils to hold the details of the encrypt/decrypt logic.
That will simplify "createUser" and "loginUser" mthods.

Reply:
done